### ISSUE TITLE
MIssing parse error on dangling `dyn`

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#19248]
```rust
fn f(_: &dyn) {}
```
does not emit a diagnostic

**Comment by Veykril** [comment]
Checking the following with the compiler
```rust
#[cfg(False)]
fn f(_: &impl) {}
#[cfg(False)]
fn f(_: &dyn) {}
```
neither emits diagnostics, so this seems to be a validation thing, so the check needs to go into https://github.com/rust-lang/rust-analyzer/blob/master/crates/syntax/src/validation.rs
